jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/AbsoluteUnixPath.java

```diff
@@ -98,6 +98,42 @@ public AbsoluteUnixPath resolve(RelativeUnixPath relativeUnixPath) {
     return new AbsoluteUnixPath(newPathComponents.build());
   }
 
+  /**
+   * Resolves this path against another relative path (by the name elements of {@code
+   * relativePath}).
+   *
+   * @param relativePath the relative path to resolve against
+   * @return a new {@link AbsoluteUnixPath} representing the resolved path
+   */
+  // TODO: Add test.
+  public AbsoluteUnixPath resolve(Path relativePath) {
+    Preconditions.checkArgument(
+        relativePath.getRoot() == null, "Cannot resolve against absolute Path: " + relativePath);
+
+    ImmutableList.Builder<String> newPathComponents =
+        ImmutableList.builderWithExpectedSize(pathComponents.size() + relativePath.getNameCount());
+    newPathComponents.addAll(pathComponents);
+    for (Path pathComponent : relativePath) {
+      if (pathComponent.toString().isEmpty()) {
+        // Skips empty components.
+        continue;
+      }
+      newPathComponents.add(pathComponent.toString());
+    }
+    return new AbsoluteUnixPath(newPathComponents.build());
```

Comment:
If going for simplicity, I guess `AbsoluteUnixPath.fromPath(Paths.get(unixPath).resolve(relativePath))` will work? The code of `fromPath()` is quite similar. (We still need to check that `relativePath` is not absolute.)